src/java_bytecode/jar_file.cpp

```diff
@@ -96,16 +138,17 @@ std::string jar_filet::get_entry(std::size_t i)
 
   std::string dest;
 
+  size_t real_index=filtered_jar[i];
```

Comment:
I'll change this to take the file name as parameter, together with a mapping from name to internal index in zip, it should be all that is needed. Having a vector of file names and using the index in that vector as entry, as it is done currently, is probably not the best way to do this.